tests/unit/s2n_record_size_test.c

```diff
@@ -140,42 +163,125 @@ int main(int argc, char **argv)
         struct s2n_connection *server_conn;
         EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
 
-        int size;
+        uint16_t size = 0;
         const int RECORD_SIZE_LESS_OVERHEADS = 1415;
 
-        EXPECT_SUCCESS(size = s2n_record_min_write_payload_size(server_conn));
+        EXPECT_OK(s2n_record_min_write_payload_size(server_conn, &size));
         EXPECT_EQUAL(RECORD_SIZE_LESS_OVERHEADS, size);
 
-        /* CBC */
-        server_conn->actual_protocol_version = S2N_TLS11;
-        server_conn->initial.cipher_suite->record_alg = &s2n_record_alg_3des_sha;
-        EXPECT_SUCCESS(size = s2n_record_min_write_payload_size(server_conn));
-        const int after_overheads = RECORD_SIZE_LESS_OVERHEADS - 20 - 8 - 1;
+        const int MIN_SIZE = RECORD_SIZE_LESS_OVERHEADS + S2N_TLS_RECORD_HEADER_LENGTH;
 
-        EXPECT_EQUAL(after_overheads - after_overheads % 8 /* rounded down to block size */
-            , size);
+        /* CBC */
+        {
+            EXPECT_SUCCESS(s2n_connection_wipe(server_conn));
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&server_conn->out));
+            server_conn->actual_protocol_version = S2N_TLS11;
+            server_conn->initial.cipher_suite->record_alg = &s2n_record_alg_3des_sha;
+            uint8_t des3_key[] = "12345678901234567890123";
+            struct s2n_blob des3 = {0};
+            EXPECT_SUCCESS(s2n_blob_init(&des3, des3_key, sizeof(des3_key)));
+            server_conn->server = &server_conn->secure;
+            EXPECT_SUCCESS(server_conn->secure.cipher_suite->record_alg->cipher->init(&server_conn->secure.server_key));
+            EXPECT_SUCCESS(server_conn->secure.cipher_suite->record_alg->cipher->init(&server_conn->secure.client_key));
+            EXPECT_SUCCESS(server_conn->secure.cipher_suite->record_alg->cipher->set_encryption_key(&server_conn->secure.server_key, &des3));
+            EXPECT_SUCCESS(server_conn->secure.cipher_suite->record_alg->cipher->set_decryption_key(&server_conn->secure.client_key, &des3));
+            EXPECT_SUCCESS(s2n_hmac_init(&server_conn->secure.server_record_mac, S2N_HMAC_SHA1, mac_key, sizeof(mac_key)));
+
+            EXPECT_OK(s2n_record_min_write_payload_size(server_conn, &size));
+            r.size = size;
+            const int after_overheads = RECORD_SIZE_LESS_OVERHEADS - RECORD_SIZE_LESS_OVERHEADS % 8; /* rounded down to cbc block size (8) */
+            #define PADDING_LENGTH_BYTE 1
+            #define RECORD_IV_SIZE 8
+            #define HMAC_DIGEST 20
```

Comment:
These should either be at the top of the file as true globals or they should be changed to const variables so that they only live in this scope. Same for your other #defines.